File: vitaes-renderer/Logger.py
import requests

def log_from_renderer(email, cv_hash, step, log_data=''):
    log(email, cv_hash, 'RENDERER', step, log_data)

def log_from_server(email, cv_hash, step, log_data=''):
    log(email, cv_hash, 'SERVER', step, log_data)

def log(email, cv_hash, origin, step, log_data):
    try:
        requests.post('http://logger:8017/', data = {
            'email': email,
            'cv_hash': cv_hash,
            'origin': origin,
            'step': step,
            'data': log_data,
        })
    except:
        pass

File: webapp/src/utils/logger.js
import fetch from 'fetch-retry';

import { getLoggerHostname } from 'utils/getHostname';

export default function logger(email, path, step, data, stacktrace) {
  const formData = new URLSearchParams();
  formData.append('email', email);
  if (path) {
    formData.append('cv_hash', path);
  }
  formData.append('origin', 'WEBAPP');
  formData.append('step', step);
  if (data) {
    formData.append('data', data);
  }
  if (stacktrace) {
    formData.append('stacktrace', stacktrace);
  }
  fetch(`${window.location.protocol}//${getLoggerHostname()}/`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
    },
    body: formData.toString(),
  });
}
